cli/azd/cmd/up.go

```diff
@@ -38,6 +43,16 @@ func upCmd(rootOptions *internal.GlobalCommandOptions) *cobra.Command {
 				},
 			),
 			&deployAction{rootOptions: rootOptions},
+			// Print the final output from infra create
+			commands.ActionFunc(
+				func(ctx context.Context, cmd *cobra.Command, args []string, azdCtx *azdcontext.AzdContext) error {
```

Comment:
Another possible design would be to create an action that both fetches the resource groups and prints the messages to the console and then make `infra create`'s command implementation a composite action that runs `infraCreateAction` and then this new `printResourceGroupsForProjectAction`, and then instead of having the action func here just have a reference to this new action.

That would let us not have to add `finalOutputRedirect` to the state of `infraCreateAction` which might be nice?

Just a thought - I feel like the `finalOutputRedirect` stuff is a little strange, but I totally understand how you ended up with it.

Reply:
Creating nested actions definitely removes the parameter, but the final design ends up being convoluted still. I think the example is why moving away from these pipelines in favor of procedural code with #625 is the way to go. I do plan on this being short-lived. I'll be making the first refactor attempt to the simplified `Action` in `up` shortly -- this is meant to be a forcing function for myself to start on that work.

I just wanted to get a bug fix out to address the reported issue, and leaving it in `provision` felt wrong as I was updating the code.

Reply:
Makes sense to me!